cmd/serv.go

```diff
@@ -282,32 +280,12 @@ func runServ(c *cli.Context) error {
 
 	//LFS token authentication
 	if verb == lfsAuthenticateVerb {
-		url := fmt.Sprintf("%s%s/%s.git/info/lfs", setting.AppURL, username, repo.Name)
-
-		now := time.Now()
-		claims := jwt.MapClaims{
-			"repo": repo.ID,
-			"op":   lfsVerb,
-			"exp":  now.Add(setting.LFS.HTTPAuthExpiry).Unix(),
-			"nbf":  now.Unix(),
-		}
-		if user != nil {
-			claims["user"] = user.ID
-		}
-		token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
-
-		// Sign and get the complete encoded token as a string using the secret
-		tokenString, err := token.SignedString(setting.LFS.JWTSecretBytes)
+		repo.OwnerName = username
+		tokenAuthentication, err := models.GenerateLFSAuthenticationToken(user, repo, lfsVerb)
```

Comment:
I think we should move `GenerateLFSAuthenticationToken` out of `models` but a sub package on `modules` maybe `generate` or `tokens`.